extensions/ql-vscode/src/data-extensions-editor/model-details/model-details-panel.ts

```diff
@@ -1,27 +1,32 @@
-import { window } from "vscode";
+import { TreeView, window } from "vscode";
 import { DisposableObject } from "../../common/disposable-object";
 import { ModelDetailsDataProvider } from "./model-details-data-provider";
-import { ExternalApiUsage } from "../external-api-usage";
 import { DatabaseItem } from "../../databases/local-databases";
+import { ExternalApiUsage, Usage } from "../external-api-usage";
 
 export class ModelDetailsPanel extends DisposableObject {
   private readonly dataProvider: ModelDetailsDataProvider;
+  private readonly treeView: TreeView<ExternalApiUsage | Usage>;
 
   public constructor() {
     super();
 
     this.dataProvider = new ModelDetailsDataProvider();
 
-    const treeView = window.createTreeView("codeQLModelDetails", {
+    this.treeView = window.createTreeView("codeQLModelDetails", {
       treeDataProvider: this.dataProvider,
     });
-    this.push(treeView);
+    this.push(this.treeView);
   }
 
   public setState(
     externalApiUsages: ExternalApiUsage[],
     databaseItem: DatabaseItem,
   ): void {
     this.dataProvider.setState(externalApiUsages, databaseItem);
+    this.treeView.badge = {
+      value: externalApiUsages.length,
+      tooltip: "Number of unmodeled external APIs",
```

Comment:
That's true, but a  problem. We should only show unmodeled APIs in this view! Shall I make the change in this PR or create a new one, since the problem wasn't introduced here?